ratis-shell/src/main/java/org/apache/ratis/shell/cli/sh/command/AbstractRatisCommand.java

```diff
@@ -215,4 +215,8 @@ protected List<RaftPeerId> getIds(String[] optionValues, BiConsumer<RaftPeerId,
     }
     return ids;
   }
+
+  protected List<RaftPeer> filterServer(List<RaftPeer> peers, RaftPeerRole role) {
+    return peers.stream().filter(it -> it.getStartupRole() == role).collect(Collectors.toList());
```

Comment:
Use startup role may not be correct since it might have changed.  Since we are already calling `getGroupManagementApi(peerId).info(groupId)` in `AbstractRatisCommand`, how about changing the `GroupInfoReply` and `RoleInfoProto` to include the listener information?

Reply:
Rewritten the code, please review it again, thanks.